base/compiler/abstractinterpretation.jl

```diff
@@ -1749,15 +1739,13 @@ function abstract_call(interp::AbstractInterpreter, arginfo::ArgInfo,
         tristate_merge!(sv, Effects()) # TODO
```

Comment:
Not quite, because we don't account for the type constraints on the opaque closure type, which implicitly get evaluated on the call and return. I can probably just go ahead and fix that while I'm at it though.